### ISSUE TITLE
Inventory not made (HTTP 500) - inventory is not more working

### ISSUE THREAD

**Issue by joschi99** [glpi-project/glpi#13990]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.6

### Bug description

Using GLPI 10.0.5, GLPI Agent 1.4 and glpiinventory 1.0.5 inventory is working fine

Migrating from GLPI 10.0.5 to 10.0.6 and glpiinventory from 1.0.5 to 1.1.0 the inventory is not more working. The agent is starting inventory, but after the inventory the data upload failed:

Apache Log:
`10.10.101.3 - - [02/Feb/2023:16:58:01 +0100] "POST /plugins/glpiinventory/ HTTP/1.1" 500 - "-" "GLPI-Agent_v1.4"`

GLPI Agent log:
`[Thu Feb  2 16:58:05 2023][error] [http client] communication error: 500 Internal Server Error, `

The computer and all other informations will not created, so no asset on inventory.

And there are a lot of errors on php-errors.log
```
[2023-02-02 16:58:03] glpiphplog.WARNING:   *** PHP User Warning (512): Invalid regular expression `GLPI Agent`. in /var/www/glpi/src/RuleCriteria.php at line 498
  Backtrace :
  src/RuleCriteria.php:498                           trigger_error()
  src/Rule.php:1714                                  RuleCriteria::match()
  src/Rule.php:1651                                  Rule->checkCriteria()
  src/Rule.php:1519                                  Rule->checkCriterias()
  src/RuleCollection.php:1626                        Rule->process()
  src/Inventory/Asset/Software.php:128               RuleCollection->processAllRules()
  src/Inventory/Inventory.php:684                    Glpi\Inventory\Asset\Software->prepare()
  src/Inventory/Inventory.php:340                    Glpi\Inventory\Inventory->processInventoryData()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:92                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  plugins/glpiinventory/front/communication.php:72   include_once()
  plugins/glpiinventory/index.php:45                 include_once()

[2023-02-02 16:58:04] glpiphplog.ERROR:   *** PHP User Error (256): Incorrect integer value: 'Tools' for column `glpi`.`glpi_softwares`.`softwarecategories_id` at row 1 in /var/www/glpi/src/DBmysql.php at line 1969
  Backtrace :
  src/DBmysql.php:1969                               trigger_error()
  src/Inventory/Asset/Software.php:714               DBmysql->executeStatement()
  src/Inventory/Asset/Software.php:443               Glpi\Inventory\Asset\Software->storeSoftware()
  src/Inventory/Asset/MainAsset.php:871              Glpi\Inventory\Asset\Software->handle()
  src/Inventory/Asset/MainAsset.php:791              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src/RuleImportAsset.php:988                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1522                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1626                        Rule->process()
  src/Inventory/Asset/MainAsset.php:571              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:706                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:341                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:92                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  plugins/glpiinventory/front/communication.php:72   include_once()
  plugins/glpiinventory/index.php:45                 include_once()

```

**Comment by JlXFjudAxpzHnK9** [comment]
I'm the same issue with 10.0.6 + GLPI agent 1.4 + glpiinventory plugin 1.1.0
Error in sql-errors.log:
`  *** MySQL query error:
  SQL: UPDATE `glpi_computers` SET `autoupdatesystems_id` = '2', `last_inventory_update` = '2023-02-03 15:18:16', `last_boot` = '2022-09-28 14:47:10', `domains_id` = 'compagny.com', `contact` = 'Administrator@WIN121352', `date_mod` = '2023-02-03 15:18:16' WHERE `id` = '2424'
  Error: Incorrect integer value: 'lohr-fr.com' for column `GLPI10`.`glpi_computers`.`domains_id` at row 1`

**Comment by stonebuzz** [comment]
Can you retry with GLPI Nighlty build ?

https://nightly.glpi-project.org/glpi/

Best regards

**Comment by GLGHus** [comment]
Hello,
 The same error 500 since GLPI 10.0.6 + GLPI agent 1.2 + 1.4.
error php : 
[2023-02-03 16:49:20] glpiphplog.ERROR:   *** PHP User Error (256): Duplicate entry 'Computer-253-19588' for key 'glpi_items_softwareversions.unicity' in C:\inetpub\wwwroot\ticket\src\DBmysql.php at line 1969
  Backtrace :
  src\DBmysql.php:1969                               trigger_error()
  src\Inventory\Asset\Software.php:909               DBmysql->executeStatement()
  src\Inventory\Asset\Software.php:481               Glpi\Inventory\Asset\Software->storeAssetLink()
  src\Inventory\Asset\MainAsset.php:871              Glpi\Inventory\Asset\Software->handle()
  src\Inventory\Asset\MainAsset.php:791              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src\RuleImportAsset.php:969                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src\Rule.php:1522                                  RuleImportAsset->executeActions()
  src\RuleCollection.php:1626                        Rule->process()
  src\Inventory\Asset\MainAsset.php:571              RuleCollection->processAllRules()
  src\Inventory\Inventory.php:706                    Glpi\Inventory\Asset\MainAsset->handle()
  src\Inventory\Inventory.php:341                    Glpi\Inventory\Inventory->handleItem()
  src\Inventory\Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src\Inventory\Request.php:90                       Glpi\Inventory\Request->inventory()
  src\Agent\Communication\AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src\Agent\Communication\AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front\inventory.php:92                             Glpi\Agent\Communication\AbstractRequest->handleRequest()

error sql : 
[2023-01-28 17:02:09] glpisqllog.ERROR: DBmysql::query() in C:\inetpub\wwwroot\ticket\src\DBmysql.php line 370
  *** MySQL query error:
  SQL: UPDATE `glpi_items_softwareversions` SET `softwareversions_id` = '6355' WHERE `id` = '432653'
  Error: Duplicate entry 'Computer-342-6355' for key 'glpi_items_softwareversions.unicity'
  Backtrace :
  src\DBmysql.php:1439                               
  src\Transfer.php:1787                              DBmysql->update()
  src\Transfer.php:1340                              Transfer->transferItemSoftwares()
  src\Transfer.php:267                               Transfer->transferItem()
  front\transfer.action.php:49                       Transfer->moveItems()
  {"user":"15@SRV-GESPARC","mem_usage":"0.001\", 4.25Mio)"} 

I was using fusion_inventory before, and i had some issues with software entity transfert duplications.

Anyway, backup my base, and dropping sofwares related tables seems to solves the issues for now (tested with 1 computer under glpi agent 1.4, will post something if an error happens in a few days)
dropped tables : 
glpi_softwarecategories        |
| glpi_softwarelicenses          |
| glpi_softwarelicensetypes      |
| glpi_softwares                 |
| glpi_softwareversions
glpi_items_softwarelicenses          |
| glpi_items_softwareversions

**BUT, i've saw that date_mod and date_creation arn't filled in 
glpi_softwareversions and glpi_softwares. (NULL)**

Hope this will help you.

**Comment by stonebuzz** [comment]
Hi @GLGHus 

can you post here related inventory file ?

Best regards

**Comment by trasher** [comment]
@GLGHus your issue is not the same as the original one here. Pleas eopen another dedicated issue with all requested information.

For the original issue, there are 2 things:
-  "Invalid regular expression `GLPI Agent`.": check and fix this rule.
- "Incorrect integer value: 'Tools' ": should be fixed in latests nightly builds.

**Comment by JlXFjudAxpzHnK9** [comment]
> Can you retry with GLPI Nighlty build ?
> 
> https://nightly.glpi-project.org/glpi/
> 
> Best regards

After update with GLPI Nighlty build (10.0-010aa5b.tar.gz). 
Always the same error:
`*** MySQL query error:
  SQL: UPDATE "glpi_computers" SET "autoupdatesystems_id" = '2', "last_inventory_update" = '2023-02-06 10:09:26', "last_boot" = '2022-09-28 14:47:10', "domains_id" = 'compagny.com', "contact" = 'Administrator@WIN1234', "date_mod" = '2023-02-06 10:09:26' WHERE "id" = '2424'
  Error: Incorrect integer value: 'compagny.com' for column "GLPI10"."glpi_computers"."domains_id" at row 1`

A new version have been released this week-end, I will retry

**Comment by trasher** [comment]
#13996 may also fix.

**Comment by JlXFjudAxpzHnK9** [comment]
After update with GLPI Nighlty build (10.0-86687fc.tar.gz)
Always the same error:
`  *** MySQL query error:
  SQL: UPDATE "glpi_computers" SET "autoupdatesystems_id" = '2', "last_inventory_update" = '2023-02-06 11:01:16', "last_boot" = '2022-09-28 14:47:10', "domains_id" = 'compagny.com', "contact" = 'Administrator@WIN1234', "date_mod" = '2023-02-06 11:01:16' WHERE "id" = '2424'
  Error: Incorrect integer value: 'compagny.com' for column "GLPI10"."glpi_computers"."domains_id" at row 1
  Backtrace :
  src/DBmysql.php:1439
  src/CommonDBTM.php:685                             DBmysql->update()
  src/CommonDBTM.php:1681                            CommonDBTM->updateInDB()
  src/Inventory/Asset/MainAsset.php:791              CommonDBTM->update()
  src/RuleImportAsset.php:969                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1519                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1625                        Rule->process()
  src/Inventory/Asset/MainAsset.php:571              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:706                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:341                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:359    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:271    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:92                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  plugins/glpiinventory/front/communication.php:72   include_once()
  plugins/glpiinventory/index.php:45                 include_once()`

**Comment by trasher** [comment]
`"glpi_computers"."domains_id"` has been removed form a while; your database is certainly not up to date. And your issiue is not the same than original one that concern sofware category.

**Comment by JlXFjudAxpzHnK9** [comment]
Understand but:
php bin/console database:update

+------------------------------------+-----------------------------------------------------+-----------------------------------------------------+
|                                    | Actuel                                              | Cible                                               |
+------------------------------------+-----------------------------------------------------+-----------------------------------------------------+
| Hôte de la base de données         | 127.0.0.1                                           |                                                     |
| Nom de la base de données          | GLPI10                                              |                                                     |
| Utilisateur de la base de données  | glpi_recette                                        |                                                     |
| Version de GLPI                    | 10.0.7-dev                                          | 10.0.7-dev                                          |
| Version de la base de données GLPI | 10.0.7-dev@21cffee0fbb5afbf0d580cabdf6fd7a922598f97 | 10.0.7-dev@21cffee0fbb5afbf0d580cabdf6fd7a922598f97 |
+------------------------------------+-----------------------------------------------------+-----------------------------------------------------+
Aucune migration requise.

So, I’ll manually modify the database.

For those who have the same mistake:
You can manually detect database inconsistencies with this command:
php bin/console database:check

regards

**Comment by cedric-anne** [comment]
This issue seems to be a mix of multiple issues that are all fixed now (fixes will be available in GLPI 10.0.7), and original requester did not give any feedback, so I close it.

If problem persists, feel free to a new issue for each distinct problem.
